pyproject.toml

```diff
@@ -24,15 +24,15 @@ urls = {Homepage = "https://jupyter.org"}
 requires-python = ">=3.9"
 dynamic = ["version"]
 dependencies = [
-    "click",
+    "click<8.2.0",
     "ghapi<=1.0.4",
     "github-activity~=0.2",
     "importlib_resources",
     "jsonschema>=3.0.1",
     "mdformat",
     "packaging",
     "pkginfo",
-    "pypiserver==2.2.0",
+    "pypiserver",
```

Comment:
We'll probably hit https://github.com/pypiserver/pypiserver/issues/630 now, let's see what CI says.

Reply:
Well, the problem is that this was only manifesting in JupyterLab as it has larger tarballs, so CI status here will not be super helpful.

Reply:
I guess we could do pins conditional on Python version? So we pin to 2.2.0 on anything older than 3.13 and allow any version on 3.13 (and it should work for most repos, but not for repos with tarballs the size of lab).

Reply:
Ah, right https://github.com/pypiserver/pypiserver/issues/630 is still open but a patch is out. So if it breaks we can now just set `PYPISERVER_BOTTLE_MEMFILE_MAX_OVERRIDE_BYTES`, perfect!

Reply:
Ah right, I thought it was initially failing in the `jupyter_releaser` repo too but then read https://github.com/jupyter-server/jupyter_releaser/issues/595 again.

yeah maybe Python pins are the way to go for now, until the next `pypiserver` release.

Reply:
I don't think it is tied to `bottle` releases as `pypiserver` is vendoring `bottle`, the patch is https://github.com/pypiserver/pypiserver/pull/636.

Now I wonder, should we set `PYPISERVER_BOTTLE_MEMFILE_MAX_OVERRIDE_BYTES` to 100 MB to match PyPI?

Reply:
> I don't think it is tied to `bottle` releases as `pypiserver` is vendoring `bottle`, the patch is [pypiserver/pypiserver#636](https://github.com/pypiserver/pypiserver/pull/636).

But this patch is not yet available in a `pypiserver` release it seems?

Reply:
Yes, It looks like there was no release since it was merged https://pypi.org/project/pypiserver/#history

Reply:
So is there anything more to be done for now until a new release is available, apart from pinning on Python versions (https://github.com/jupyter-server/jupyter_releaser/pull/607/commits/aada747b57352a44c2376d43da587a0da902e738)?

Reply:
No, I think we can merge :)